src/mon/MgrStatMonitor.cc

```diff
@@ -119,6 +130,18 @@ void MgrStatMonitor::clear_pool_availability(int64_t poolid)
   dout(20) << __func__ << " cleared availability score for pool: " << poolid << dendl;
 }
 
+bool MgrStatMonitor::should_calc_pool_availability() 
+{
+  dout(20) << __func__ << dendl;
+  std::scoped_lock l(lock); 
+
+  utime_t now = ceph_clock_now();
+  if ((now  - pool_availability_last_updated) >= pool_availability_update_interval) {
```

Comment:
`pool_availability_last_updated` I don't see this getting updated anywhere after `calc_pool_availability()` finishes